lightclient/src/platform/wasm_helpers.rs

```diff
@@ -32,379 +32,11 @@ pub fn sleep(duration: Duration) -> Delay {
-        read_buffer_reasonable_capacity: usize,
-        write_buffers: &'a mut Vec<Vec<u8>>,
-        write_closed: &'a mut bool,
-        close_pending: &'a mut bool,
-        read_write_wake_up_after: &'a mut Option<Instant>,
-    }
-
-    impl<'a> ops::Deref for ReadWriteAccess<'a> {
-        type Target = read_write::ReadWrite<Instant>;
-
-        fn deref(&self) -> &Self::Target {
-            &self.read_write
-        }
-    }
-
-    impl<'a> ops::DerefMut for ReadWriteAccess<'a> {
-        fn deref_mut(&mut self) -> &mut Self::Target {
-            &mut self.read_write
-        }
-    }
-
-    impl<'a> Drop for ReadWriteAccess<'a> {
-        fn drop(&mut self) {
-            *self.read_buffer = mem::take(&mut self.read_write.incoming_buffer);
-            *self.read_buffer_valid = self.read_buffer.len();
-
-            // Adjust `read_buffer` to the number of bytes requested by the consumer.
-            if let Some(expected_incoming_bytes) = self.read_write.expected_incoming_bytes {
-                if expected_incoming_bytes < self.read_buffer_reasonable_capacity
-                    && self.read_buffer.is_empty()
-                {
-                    // We use `shrink_to(0)` then `reserve(cap)` rather than just `shrink_to(cap)`
-                    // so that the `Vec` doesn't try to preserve the data in the read buffer.
-                    self.read_buffer.shrink_to(0);
-                    self.read_buffer
-                        .reserve(self.read_buffer_reasonable_capacity);
-                } else if expected_incoming_bytes > self.read_buffer.len() {
-                    self.read_buffer
-                        .reserve(expected_incoming_bytes - self.read_buffer.len());
-                }
-                debug_assert!(self.read_buffer.capacity() >= expected_incoming_bytes);
-            }
-
-            *self.write_buffers = mem::take(&mut self.read_write.write_buffers);
-
-            if self.read_write.write_bytes_queueable.is_none() && !*self.write_closed {
-                *self.write_closed = true;
-                *self.close_pending = true;
-            }
-
-            *self.read_write_wake_up_after = self.read_write.wake_up_after.take();
-
-            // If the consumer has advanced its reading or writing sides, we make the next call to
-            // `read_write_access` return immediately by setting `wake_up_after`.
-            if (self.read_buffer_len_before != self.read_buffer.len()
-                && self
-                    .read_write
-                    .expected_incoming_bytes
-                    .map_or(false, |b| b <= self.read_buffer.len()))
-                || (self.write_buffers_len_before != self.write_buffers.len()
-                    && !*self.write_closed)
-            {
-                *self.read_write_wake_up_after = Some(self.read_write.now);
-            }
-        }
-    }
-}
+pub struct Stream(
+    #[pin]
+    pub  smoldot::libp2p::with_buffers::WithBuffers<
```

Comment:
```suggestion
    pub smoldot::libp2p::with_buffers::WithBuffers<
```

Reply:
I also found this annoying, however it seems this is how rustfmt is formatting this for me 🤔

Reply:
That's very strange; I haven't seen such double-space formatting before; moving to single space casues a fmt error?

Maybe the macro leads to something weird hmmm

Reply:
Yep that's super weird:

```bash
cargo fmt --all -- --check                                                              19:42:04
Diff in /home/lexnv/workspace/subxt/lightclient/src/platform/wasm_helpers.rs at line 34:
 #[pin_project::pin_project]
 pub struct Stream(
     #[pin]
-    pub smoldot::libp2p::with_buffers::WithBuffers<
+    pub  smoldot::libp2p::with_buffers::WithBuffers<
         future::BoxFuture<'static, Result<WasmSocket, std::io::Error>>,
         WasmSocket,
         Instant,
```

And also the CI fmt check is passing: https://github.com/paritytech/subxt/actions/runs/7103856224/job/19337498624?pr=1307